sound/soc/sof/intel/mtl.c

```diff
@@ -493,7 +493,8 @@ int mtl_dsp_cl_init(struct snd_sof_dev *sdev, int stream_tag, bool imr_boot)
 	return 0;
 
 err:
-	snd_sof_dsp_dbg_dump(sdev, "MTL DSP init fail", 0);
+	if (hda->boot_iteration == HDA_FW_BOOT_ATTEMPTS)
+		snd_sof_dsp_dbg_dump(sdev, "MTL DSP init fail", 0);
```

Comment:
it should be done in a same way as in hda-loader.c:
https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda-loader.c#L207C4-L218
```c
	flags = SOF_DBG_DUMP_PCI | SOF_DBG_DUMP_MBOX | SOF_DBG_DUMP_OPTIONAL;

	/* after max boot attempts make sure that the dump is printed */
	if (hda->boot_iteration == HDA_FW_BOOT_ATTEMPTS)
		flags &= ~SOF_DBG_DUMP_OPTIONAL;

	dump_msg = kasprintf(GFP_KERNEL, "Boot iteration failed: %d/%d",
			     hda->boot_iteration, HDA_FW_BOOT_ATTEMPTS);
	snd_sof_dsp_dbg_dump(sdev, dump_msg, flags);
```

Reply:
Ack, I feel this is a better way cause once seeing snd_sof_dsp_dbg_dump(), we either ignore it or enable the debug print at  previous steps.